### PR TITLE
[rewards p2] Use tax rate when applying tx

### PR DESCRIPTION
**Pull request by ninjaahhh**
plus some naming / coding style changes

two relevant parts:

1. when applying transactions, fees are multiplied with the reward tax rate (e.g. 0.5)
2. when collecting x-shard tx fees, multiply with the same tax rate

comparing with previous approach (processing fee as usual, then deduct half from coinbase address when finalizing the block), this should reflect gas dynamics more accurately

### CODE REVIEW

File: quarkchain/cluster/shard_state.py [hunk 1]
```diff
@@ -556,10 +558,6 @@ def run_block(
                 )
                 raise e
 
-        # Put only half of block fee to coinbase address
-        check(evm_state.get_balance(evm_state.block_coinbase) >= evm_state.block_fee)
-        evm_state.delta_balance(evm_state.block_coinbase, -evm_state.block_fee // 2)
```
Comment (ninjaahhh) [inline quarkchain/cluster/shard_state.py]: moved this part of logic to 1) applying tx and 2) collecting x-shard fee, respectively

File: quarkchain/cluster/tests/test_shard_state.py [hunk 2]
```diff
@@ -1181,8 +1182,11 @@ def test_add_block_receipt_root_not_match(self):
         # Should succeed
         state.finalize_and_add_block(b1)
         b1.finalize(evm_state=state.run_block(b1))
-        b1.meta.hash_evm_receipt_root = b"00" * 32
-        self.assertRaises(ValueError, state.add_block(b1))
```
Comment (ninjaahhh) [inline quarkchain/cluster/tests/test_shard_state.py]: this won't assert raise, need to be used as context manager

File: quarkchain/cluster/tests/test_utils.py [hunk 3]
```diff
@@ -61,6 +61,11 @@ def get_test_env(
     env.quark_chain_config.SKIP_ROOT_DIFFICULTY_CHECK = True
     env.cluster_config.ENABLE_TRANSACTION_HISTORY = True
     env.cluster_config.DB_PATH_ROOT = ""
+    # prent breaking previous tests after tweaking default rewards
```
Comment (qcgg) [inline quarkchain/cluster/tests/test_utils.py]: # prevent?

File: quarkchain/config.py [hunk 4]
```diff
@@ -191,7 +191,7 @@ class RootConfig(BaseConfig):
     GENESIS = None  # type: RootGenesis
 
     COINBASE_ADDRESS = bytes(24).hex()
-    COINBASE_AMOUNT = 5
+    COINBASE_AMOUNT = 120 * QUARKSH_TO_JIAOZI
```
Comment (qcgg) [inline quarkchain/config.py]: shall we calculate this based on other parameters?

**Pull request merged**